plutus-ledger-api/src/PlutusLedgerApi/Common/ParamName.hs

```diff
@@ -59,15 +60,19 @@ instance (GIsParamName a, GIsParamName b) => GIsParamName ((:+:) a b) where
     gshowParamName (R1 x) = gshowParamName x
 
 -- Given an ordered list of parameter values, tag them with their parameter names.
-tagWithParamNames :: forall k m. (Enum k, Bounded k, MonadError CostModelApplyError m) => [Integer] -> m [(k, Integer)]
+tagWithParamNames :: forall k m. (Enum k, Bounded k, MonadError CostModelApplyError m, MonadWriter [CostModelApplyWarn] m)
```

Comment:
I know that:
1) MonadWriter.CPS is better than MonadWriter.Strict but it is not in our mtl version 2.2.2
2) using list [] as the log datatype is worse than others (dlist/endo) but i thought it does not matter much here

Reply:
Yeah, I think it's fine since we're `tell`-ing at most one thing! Worth recording your reasoning though